modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -764,7 +734,7 @@ private void checkSefReferences(DiagnosticPos pos, SymbolEnv env, BVarSymbol var
 
     private void setExprType(BLangExpression expr, List<BType> expTypes) {
         int expected = expTypes.size();
-        if (expr.isMultiReturnExpr()) {
+        if (expr instanceof MultiReturnExpr) {
```

Comment:
can't we use kind?

Reply:
we can't use Kind here, as there are multiple kinds are associated.